test/unit/client/test_client.py

```diff
@@ -481,6 +483,94 @@ def test_create_enterprise_user_returns_the_correct_user_object(mock_client, moc
+    expected_data = {
+        'policy_name': policy_name,
+        'policy_type': policy_type,
+        'retention_length': 5,
+        'disposition_action': disposition_action,
+    }
+    mock_policy = {
+        'type': 'retention_policy',
+        'id': '1234',
+        'policy_name': policy_name,
+    }
+    mock_box_session.post.return_value.json.return_value = mock_policy
+    policy = mock_client.create_retention_policy(policy_name, disposition_action, 5)
+    mock_box_session.post.assert_called_once_with(expected_url, data=json.dumps(expected_data))
+    assert policy.object_id == mock_policy['id']
+    assert policy.object_type == mock_policy['type']
+    assert policy.policy_name == policy_name
+    assert isinstance(policy, RetentionPolicy)
+
+
+def test_create_infinte_retention_policy(mock_client, mock_box_session):
+    policy_name = 'Test Retention Policy'
+    policy_type = 'indefinite'
+    disposition_action = 'remove_retention'
+    expected_url = "{0}/retention_policies".format(API.BASE_API_URL)
+    expected_data = {
+        'policy_name': policy_name,
+        'policy_type': policy_type,
+        'disposition_action': disposition_action,
+    }
+    mock_policy = {
+        'type': 'retention_policy',
+        'id': '1234',
+        'policy_name': policy_name,
+    }
+    mock_box_session.post.return_value.json.return_value = mock_policy
+    policy = mock_client.create_retention_policy(policy_name, disposition_action, float('inf'))
+    mock_box_session.post.assert_called_once_with(expected_url, data=json.dumps(expected_data))
+    assert policy.object_id == mock_policy['id']
+    assert policy.object_type == mock_policy['type']
+    assert policy.policy_name == policy_name
+    assert isinstance(policy, RetentionPolicy)
+
+
+def test_get_retention_policies(mock_client, mock_box_session):
+    expected_url = "{0}/retention_policies".format(API.BASE_API_URL)
+    mock_policy = {
+        'type': 'retention_policy',
+        'id': '12345',
+        'name': 'Test Retention Policy',
+    }
+    mock_box_session.get.return_value.json.return_value = {
+        'limit': 100,
+        'entries': [mock_policy],
+        'next_marker': 'testMarker',
+    }
+    policies = mock_client.get_retention_policies()
+    policy = policies.next()
+    mock_box_session.get.assert_called_once_with(expected_url, params={})
+    assert isinstance(policy, RetentionPolicy)
+    assert policy.id == mock_policy['id']
+    assert policy.name == mock_policy['name']
+
+
+def test_get_file_version_retentions(mock_client, mock_box_session):
+    expected_url = "{0}/file_version_retentions".format(API.BASE_API_URL)
+    mock_retention = {
+        'type': 'file_version_retention',
+        'id': '12345',
+    }
+    mock_box_session.get.return_value.json.return_value = {
+        'limit': 100,
+        'entries': [mock_retention],
+        'next_marker': 'testMarker',
+    }
+    retentions = mock_client.get_file_version_retentions()
```

Comment:
This method in particular has a bunch of optional params, none of which are covered by this test.  All of those branches could be working incorrectly and we'd never know